File: BE/src/main/java/kr/codesquad/issuetracker/web/dto/label/LabelDto.java
package kr.codesquad.issuetracker.web.dto.label;

import kr.codesquad.issuetracker.domain.label.Color;
import kr.codesquad.issuetracker.domain.label.Label;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

@Getter
@AllArgsConstructor
public class LabelDto {

	private Long id;
	private	String title;
	private String description;
	private Color color;

	public static LabelDto from(Label label) {
		return new LabelDto(
			label.getId(),
			label.getTitle(),
			label.getDescription(),
			label.getColor()
		);
	}
}

File: app/src/main/java/com/example/it/issuetracker/data/dto/LabelDto.kt
package com.example.it.issuetracker.data.dto

import com.example.it.issuetracker.domain.model.Label
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class LabelDto(
    val id: Int,
    val title: String,
    val description: String,
    val color: String,
    val textColor: String,
)

fun LabelDto.toLabel(): Label =
    Label(id = id, title = title, description = description, color = color, textColor = textColor)